### ISSUE TITLE
inventory clears itself(?)

### ISSUE THREAD

**Issue by jrufus09** [bryleinandan/MPGD-Main-Game#9]
Issue is mainly with the InventorySlots[] sometimes deciding its length is zero, then other times remembering its length is 32 -- within InventoryManager.cs -- resulting in index out of bounds error when scrolling through the inventory

this happens despite initialising InventorySlots[] with a length of 32

perhaps there is another InventorySlot[] arr somewhere? on quick inspection, I can't see any code that alters the length of the array

**Comment by jrufus09** [comment]
https://discussions.unity.com/t/array-length-is-zero/575894/2

this sort of fixed the problem by setting InventorySlots[] to private: the array now agrees it's always 32 items big. index out of bounds is still happening.

**Comment by jrufus09** [comment]
my bad it is now a nullreference error lemme fix

**Comment by jrufus09** [comment]
https://discussions.unity.com/t/public-arrays-reset-themselves/184563/2

internal solved the solution in the same way.

also i think the nullobject error it throws now is because there's a function (Deselect(), Select()) being called on a null item (there's no object in the array) so I suspect internal came back to bite me: the inspector is what assigns objects into this array to begin with.

though this happens with both private [] and internal[] i fear we may have to assign the 31 objects through code

**Comment by jrufus09** [comment]
either we manually instanciate the 32 objects at runtime or, if they already exist in the inspector,
- give them all a tag
- "get all objects by tag" and throw them into an array
- order them? use this array to reference the inventory you need

**Comment by jrufus09** [comment]
commiting code purely for the six lines of debug i added

**Comment by jrufus09** [comment]
renderers refused to actually disable so i just made the ui scale to 0. this lets main inventory be active all the time without being visible.

so that means all the item objects (all 32 of them) can remain active in the game: I tagged them all, and grab everything with that tag when InventoryManager starts. then we throw it into the original array.

manually populating the inventory through the code means the unity editor doesn't do strange things with overwriting the array we defined (which i worked out was happening since the length sometimes becomes zero despite no changes having been made anywhere else in the code, and why the length of inventorySlots was constant when I made it private or internal, but sometimes became 0 when public: unity's editor is allowed to change the value of public variables and overwrite with whatever is in its editor)
